### PR TITLE
Fix issues around JWT updated propagation to Redux

### PR DESCRIPTION
**Pull request by paweldomas**

### CODE REVIEW

File: spot-client/src/common/backend/SpotBackendService.test.js [hunk 1]
```diff
@@ -155,6 +167,20 @@ describe('SpotBackendService', () => {
                             .toBe(true);
                     });
             });
+            it('notifies the listeners of JWT update - when starting with the registration expired', () => {
+                dateSpy = jest.spyOn(Date, 'now');
+                dateSpy.mockReturnValue(Date.now() + MOCK_RESPONSE.expiresIn);
```
Comment (virtuacoplenny) [inline spot-client/src/common/backend/SpotBackendService.test.js]: Oh man what is this? I've been using lolex. This looks better.

File: spot-client/src/common/backend/SpotBackendService.test.js [hunk 2]
```diff
@@ -210,6 +236,43 @@ describe('SpotBackendService', () => {
...
+                            mucUrl: 'muc-url',
+                            name: 'mock-muc-name'
+                        }));
+
+                        return spotBackendService.getRoomInfo().then(roomInfo => {
+                            expect(roomInfo).toEqual({
+                                id: 'mock-id',
+                                name: 'mock-muc-name',
+                                roomName: 'muc-url'
+                            });
```
Comment (virtuacoplenny) [inline spot-client/src/common/backend/SpotBackendService.test.js]: I see a chunk of duplication with the above. Do you think it's worthwhile to move this into a helper?

File: spot-client/src/common/backend/SpotBackendService.js [hunk 3]
```diff
@@ -133,9 +133,14 @@ export class SpotBackendService extends Emitter {
      */
     _maybeClearRegistration(error) {
         if (this.isUnrecoverableRequestError(error)) {
-            this.getJwt() && logger.log('Cleared backend registration');
-            persistence.set(PERSISTENCE_KEY, undefined);
-            this.registration = undefined;
+            if (this.registration) {
+                logger.log('Cleared backend registration');
```
Comment (virtuacoplenny) [inline spot-client/src/common/backend/SpotBackendService.js]: Great move. So obvious this should have been this way.

File: spot-client/src/common/backend/SpotBackendService.js [hunk 4]
```diff
@@ -149,6 +154,10 @@ export class SpotBackendService extends Emitter {
      * @private
      */
     _refreshRegistration(registration) {
+        if (!registration) {
+            throw new Error('No registration object passed to _refreshRegistration');
```
[the remark this answers is not in the document]
Comment (paweldomas) [reply spot-client/src/common/backend/SpotBackendService.js]: right, it should be a rejected promise

File: spot-client/src/common/backend/SpotBackendService.js [hunk 5]
```diff
@@ -225,6 +229,12 @@ export class SpotBackendService extends Emitter {
             + `scheduling refresh to be done in ${delay / ONE_MINUTE} minutes`);
 
         this._refreshTimeout = setTimeout(() => {
+            if (!this.registration) {
+                logger.error('Unable to refresh access token - no registration');
+
+                return;
```
[the remark this answers is not in the document]
Comment (paweldomas) [reply spot-client/src/common/backend/SpotBackendService.js]: If the logic from this commit is ok, then the event would have been emitted whenever this.registration was set to undefined.

File: spot-client/src/common/backend/SpotBackendService.test.js (hunk 2 above)
Comment (paweldomas) [reply spot-client/src/common/backend/SpotBackendService.test.js]: right, will do

File: spot-client/src/common/backend/SpotBackendService.test.js (hunk 1 above)
Comment (paweldomas) [reply spot-client/src/common/backend/SpotBackendService.test.js]: funny I was trying to find lolex usage, but was not able to find it :P

File: spot-client/src/common/backend/SpotBackendService.test.js (hunk 2 above)
Comment (paweldomas) [reply spot-client/src/common/backend/SpotBackendService.test.js]: Okay I gave it a try, please let me know if that's what you had in mind. It's in separate commit and eventually could be left this way or I can squash and rebase if you want me to.

**Pull request merged**